Patch makefile to build liblpeg.a for wasm.

Also patch sources to make available function to load lpeg from Lua.

--- a/makefile
+++ b/makefile
@@ -1,5 +1,5 @@
 LIBNAME = lpeg
-LUADIR = ./lua/
+LUADIR ?= ./lua/
 
 COPT = -O2 -DNDEBUG
 # COPT = -O0 -DLPEG_DEBUG -g
@@ -26,10 +26,13 @@
 
 
 CFLAGS = $(CWARNS) $(COPT) -std=c99 -I$(LUADIR) -fPIC
-CC = gcc
+CC = pushd $(dir ${EMSDK_SHELL}) >&/dev/null && source ${EMSDK_SHELL} >&/dev/null && popd >&/dev/null && emcc
 
 FILES = lpvm.o lpcap.o lptree.o lpcode.o lpprint.o lpcset.o
 
+liblpeg.a: $(FILES)
+	pushd $(dir ${EMSDK_SHELL}) >&/dev/null && source ${EMSDK_SHELL} >&/dev/null && popd >&/dev/null && emar rcu $@ $(FILES)
+
 # For Linux
 linux:
 	$(MAKE) lpeg.so "DLLFLAGS = -shared -fPIC"
--- /dev/null	2025-08-18 10:45:23.895217163 +0900
+++ b/lpeg_exported.h	2025-08-18 19:57:38.195314700 +0900
@@ -0,0 +1,14 @@
+#ifndef LPEG_FOR_WASM_LPEG_EXPORTED_H_
+#define LPEG_FOR_WASM_LPEG_EXPORTED_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern int luaopen_lpeg (lua_State *L);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
--- a/lptree.c	2025-08-18 19:55:14.459573646 +0900
+++ b/lptree.c	2025-08-18 19:55:35.497907284 +0900
@@ -14,6 +14,7 @@
 #include "lptree.h"
 #include "lpcset.h"
 
+#include "lpeg_exported.h"
 
 /* number of siblings for each tree */
 const byte numsiblings[] = {
@@ -1382,7 +1383,6 @@
 };
 
 
-int luaopen_lpeg (lua_State *L);
 int luaopen_lpeg (lua_State *L) {
   luaL_newmetatable(L, PATTERN_T);
   lua_pushnumber(L, MAXBACK);  /* initialize maximum backtracking */
